calyx-ir/src/structure.rs

```diff
@@ -479,6 +479,18 @@ impl<T> Assignment<T> {
     }
 }
 
+impl From<&Assignment<Nothing>> for Assignment<StaticTiming> {
```

Comment:
Don't implement `From<&T>` since it hides the fact that there is a clone happening.